core/storage/topic/gae_models.py

```diff
@@ -388,6 +383,15 @@ class TopicSummaryModel(base_models.BaseModel):
     # The thumbnail background color of the topic.
     thumbnail_bg_color = datastore_services.StringProperty(indexed=True)
     version = datastore_services.IntegerProperty(required=True)
+    # A mapping from each story id belonging to the topic to a list of
+    # exploration ids that are linked to the corresponding published story. This
+    # mapping represents the relationships between a topic, its published
+    # stories and the explorations that are linked to each story. This field
```

Comment:
I just realized that, with the [serial chapter launch feature](https://docs.google.com/document/d/1Uh_CJRhGE4IM7c0ENdmXkoMD0YtGqFsl1ObR6vvmxdY/edit#heading=h.liiqbmwmz8j8) implemented by @Kej-r03, chapters (or rather, story nodes) will have a publication status. So when we talked about "published" we need to be clear whether this means "all chapters in a published story" or "published chapters in a published story" (though the latter concept only comes into play when that serial chapters feature is launched, it's behind a feature flag right now).

In terms of what we want to show in the CD, it's probably any chapters that are published or queued for publication, since those are likely to be final and their content shouldn't change much. I'm not sure how hard that is to work into the current PR, though. It's probably not a big problem if we make all chapters (including draft) available for now, but if so let's file an issue to change this to the condition mentioned above, and flag the relevant parts of the code that need to be changed. Thanks!

Reply:
When you said any chapters that are "queued for publication", is that the same as any story node with status "Ready for publishing"? If so, then I think it be better to show only chapters that are published. I don't see the point of giving the chapter owner the choice to publicize said chapter if the app will implicitly publicize the chapter to contributors.

Reply:
Re your question in the first sentence, yes that's correct. I think what you suggest is fine too -- the lesson creator is mainly thinking about publication to learners and it's really a question of whether we want translators to have an "advance preview". But given that the lesson can still be changed prior to publication, it's probably safer to publish it for all audiences at once, as you suggest.